File: cycling_power.py
#!/usr/bin/env python3
from argparse import ArgumentParser


def main(ftp, weight):
    norm_ftp = ftp / 100

    print(f"Power zones at {ftp:.0f}W FTP")
    print(f"    Recovery       {norm_ftp * 0:3.0f} to {norm_ftp * 55:3.0f}W")
    print(f"    Endurance      {norm_ftp * 55:3.0f} to {norm_ftp * 75:3.0f}W")
    print(f"    Sweetspot      {norm_ftp * 75:3.0f} to {norm_ftp * 90:3.0f}W")
    print(f"    Threshold      {norm_ftp * 90:3.0f} to {norm_ftp * 105:3.0f}W")
    print(f"    VO2Max         {norm_ftp * 105:3.0f} to {norm_ftp * 120:3.0f}W")
    print(f"    Anaerobic      {norm_ftp * 120:3.0f} to {norm_ftp * 150:3.0f}W")
    print(f"    Neuromuscular  >{norm_ftp * 150:3.0f}W")

    print(f"\nTheoretical W/kg ")
    print(f"    NOW {ftp / weight:.1f} W/kg")
    print(f"    for 4   W/kg, need {4 * weight:.0f} W")
    print(f"    for 3.5 W/kg, need {3.5 * weight:.0f} W")
    print(f"    for 3   W/kg, need {3 * weight:.0f} W")


if __name__ == "__main__":
    parser = ArgumentParser("cycling power")
    parser.add_argument("--ftp", type=float, required=True)
    parser.add_argument("--weight", type=float, help="Weight in kg", required=True)
    args = parser.parse_args()

    main(args.ftp, args.weight)

File: cyclingpower.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	ftp := 259.0
	ftp_norm := ftp / 100.0
	weight := 74.0

	fmt.Printf("Cycling Power Zones at %.0fW FTP\n", ftp)
	fmt.Printf("    Recovery      %3.0f to %3.0fW\n", ftp_norm*0.0, ftp_norm*55.0)
	fmt.Printf("    Endurance     %3.0f to %3.0fW\n", ftp_norm*55.0, ftp_norm*75.0)
	fmt.Printf("    Sweetspot     %3.0f to %3.0fW\n", ftp_norm*75.0, ftp_norm*90.0)
	fmt.Printf("    Threshold     %3.0f to %3.0fW\n", ftp_norm*90.0, ftp_norm*105.0)
	fmt.Printf("    VO2Max        %3.0f to %3.0fW\n", ftp_norm*105.0, ftp_norm*120.0)
	fmt.Printf("    Anaerobic     %3.0f to %3.0fW\n", ftp_norm*120.0, ftp_norm*150.0)
	fmt.Printf("    Neuromuscular >%vW\n", ftp_norm*150/100)
	fmt.Printf("\nTheoretical W/kg\n")
	fmt.Printf("    NOW %.1f W/kg\n", ftp/weight)
	fmt.Printf("    for 4   W/kg, %.0fW\n", 4*weight)
	fmt.Printf("    for 3.5 W/kg, %.0fW\n", 3.5*weight)
	fmt.Printf("    for 3   W/kg, %.0fW\n", 3*weight)
	strings.Join([]string{}, "")
}
